### PR TITLE
[XEDRA Evolved] Add ARMOR_ALL enchantment and use it in two new Homullus mutations

### PR DESCRIPTION
**Pull request by b3brodie**
#### Summary
Mods "[XEDRA Evolved] Add ARMOR_ALL enchantment and use it in two new Homullus mutations"

#### Purpose of change
Homullus has / is planned to have mutations around crafting new tools or buildings.  However, using tools and equipment is also an important part of human existence which deserves to be reflected.  This PR adds a pre and post threshold mutation that improves armor values for homullus's.

#### Describe the solution
ARMOR_ALL enchantment: Universal armor enchantment, which should simplify writing out each separate armor enchantment value.  It should also support custom damage types not yet covered by the existing specific armor type enchantments.

Reinforced Armor: Increases armor values by 20%.  Rather than giving a flat armor increase, this mutation increases the strength of armor that homullus's wear, adding more benefit to a suit of armor compared to a heavy shirt.  I feel that this fits better with enhanced tool use that flat armor increases.

Fortified Armor: Increases armor values by 40%.  Post Threshold.

#### Describe alternatives you've considered
I've thought up a couple more ways to improve tool use for Homullus's, but each of them requires actual code changes so I'm planning on doing them in separate PRs.

#### Testing
Equipped a kevlar jumpsuit on a character and let a tough zombie attack me, observing that I was damaged.
Created a test mutation with 5x armor_all and let a tough zombie attack me, observing 0 damage taken at all. (ignoring head).

### CODE REVIEW

File: doc/MAGIC.md [hunk 1]
```diff
@@ -922,7 +923,8 @@ A small subset of enchantments can be applied to monsters via effects. These are
 
 Character status value  | Description
 ---                     |---
-`ARMOR_ACID`            | Negative values give armor against the damage, positive values make the monster accept more damage of this type.
+`ARMOR_ALL`             | Negative values give armor against the damage, positive values make the monster accept more damage of this type.
```
Comment (RenechCDDA) [inline doc/MAGIC.md:926]: ARMOR_ALL actually comes after ARMOR_ACID alphabetically.

File: src/character_armor.cpp [hunk 2]
```diff
@@ -119,6 +119,7 @@ static void armor_enchantment_adjust( Character &guy, damage_unit &du )
     } else if( du.type == STATIC( damage_type_id( "bullet" ) ) ) {
         du.amount = guy.calculate_by_enchantment( du.amount, enchant_vals::mod::ARMOR_BULLET );
     }
+    du.amount = guy.calculate_by_enchantment( du.amount, enchant_vals::mod::ARMOR_ALL );
```
Comment (RenechCDDA) [inline src/character_armor.cpp:122]: This breaks the `pure` damage type (mostly found in magiclysm). Pure is meant to never be reduced.

Recommend you check if damage type is NOT pure and then apply ARMOR_ALL's reduction.

File: doc/MAGIC.md (hunk 1 above)
Comment (b3brodie) [reply doc/MAGIC.md]: Alphabetized this and magic_enchant.h

File: src/character_armor.cpp (hunk 2 above)
Comment (b3brodie) [reply src/character_armor.cpp]: I forgot about pure damage.  Added a check to not add ARMOR_ALL if its pure damage.

**Pull request merged**